---
# Documentation: https://wowchemy.com/docs/managing-content/

title: "Menggunakan Facebook Prophet untuk meramal harga beras"
subtitle: ""
summary: ""
authors: [admin] 
tags: [perdagangan, ekonomi]
categories: []
date: 2025-08-02T11:00:00+07:00
lastmod: 2025-08-02T11:00:00+07:00
featured: false
draft: false

# Featured image
# To use, add an image named `featured.jpg/png` to your page's folder.
# Focal points: Smart, Center, TopLeft, Top, TopRight, Left, Right, BottomLeft, Bottom, BottomRight.
image:
  caption: ""
  focal_point: ""
  preview_only: false

# Projects (optional).
#   Associate this post with one or more of your projects.
#   Simply enter your project's folder or file name without extension.
#   E.g. `projects = ["internal-project"]` references `content/project/deep-learning/index.md`.
#   Otherwise, set `projects = []`.
projects: []
---



In [3]:
# Bikin class dulu
# Optimisasi sudah dilakukan dengan tangan di atas, jadi class-nya ga perlu optimisasi lagi. tinggal kalkulasi biasa.
import pandas as pd
import numpy as np

class ekon:
  r"""
  
  toy model of a closed economy with fixed number of labor, fixed productivity, 2 goods A and B, 1 factor of production.

  

  """
  
  def __init__(self, L=200, ## kasih nilai default untuk 6 parameter eksogen
                     f=0.25,
                     g=0.75,
                     alpha=0.5,
                     beta=0.5,
                     w=1,

                     
                     ):
    self.L,self.f,self.g,self.alpha,self.beta,self.w=L,f,g,alpha,beta,w
    
  def hasil(self):
    L,f,g,alpha,beta,w=self.L,self.f,self.g,self.alpha,self.beta,self.w
    LA=alpha/(alpha+beta)*L
    LB=beta/(alpha+beta)*L
    A=f*LA
    B=g*LB
    U=(A**alpha)*(B**beta)
    pa=w/f
    pb=w/g
    p=pa*(A/(A+B))+pb*(B/(A+B))
    npdb=pa*A+pb*B
    pdb=npdb/p
    return print(f'Goods market: \n harga A adalah {pa:,.2f} dan harga B adalah {pb:,.2f} \n Jumlah produksi (A,B) adalah {(A,B)} \nLabor market: \n gaji adalah {w:,.2f} \n dari total {L:,.2f} pekerja, sebanyak {LA:,.2f} bekerja di industri A dan {LB:,.2f} di industri B \nmakro: \n PDB nominal adalah {npdb:,.2f} dan {npdb/L:,.2f} per kapita  \n PDB riil adalah {pdb:,.2f} dan {pdb/L:,.2f} per kapita \n level harga adalah {p:,.2f} \n welfare di negara ini adalah {U:,.2f} dan {U/L:,.2f} per kapita ')

Kita tes modelnya yuk. Pertama, run kode di atas, lalu run default parameters dengan cara di bawah ini.

In [4]:
ekon().hasil()

Goods market: 
 harga A adalah 4.00 dan harga B adalah 1.33 
 Jumlah produksi (A,B) adalah (25.0, 75.0) 
Labor market: 
 gaji adalah 1.00 
 dari total 200.00 pekerja, sebanyak 100.00 bekerja di industri A dan 100.00 di industri B 
makro: 
 PDB nominal adalah 200.00 dan 1.00 per kapita  
 PDB riil adalah 100.00 dan 0.50 per kapita 
 level harga adalah 2.00 
 welfare di negara ini adalah 43.30 dan 0.22 per kapita 


Sesuai prediksi, harga A (pangan) lebih tinggi daripada harga B (sandang) karena ekonomi negara ini lebih produktif membuat sandang daripada pangan.

Tentu saja kita bisa coba pakai angka lain. Let's say kita punya negara lain dengan karakteristik jumlah tenaga kerja 400 juta orang dan relatif lebih produktif agrikulturnya daripada pakaian jadi-nya. Oh, kita juga bisa misalkan wage level di negara ini lebih tinggi 3x lipat daripada negara sebelah.

In [5]:
ekon(L=400,w=3,f=0.8,g=0.2).hasil()

Goods market: 
 harga A adalah 3.75 dan harga B adalah 15.00 
 Jumlah produksi (A,B) adalah (160.0, 40.0) 
Labor market: 
 gaji adalah 3.00 
 dari total 400.00 pekerja, sebanyak 200.00 bekerja di industri A dan 200.00 di industri B 
makro: 
 PDB nominal adalah 1,200.00 dan 3.00 per kapita  
 PDB riil adalah 200.00 dan 0.50 per kapita 
 level harga adalah 6.00 
 welfare di negara ini adalah 80.00 dan 0.20 per kapita 


Katakanlah negara pertama kita sebut Home (H) sementara negara ke-2 kita sebut Foreign (F). Ciri utama adalah negara F lebih besar (400 pekerja dibanding 200) dan lebih produktif di agrikultur ketimbang garmen. Kita langsung lihat bahwa negara H membuat A=25 dan B=75, lebih condong banyak bikin B. Sementara itu, negara F lebih berat ke A, membuat A sebanyak 160 sementara B sebanyak 40, atau perbedaan 4 kali lipat.

Kita juga bisa bereksperimen dengan karakteristik lain, misalnya negara F lebih suka garmen sehingga $\beta=0.7$ sementara $\alpha=0.3$, dan lain sebagainya.

Nah pertanyaan lebih menariknya adalah, apa yang terjadi kalau kedua negara ini melakukan trade? Ntar ya saya bikinin lagi kalo postingan ini rame wkwk.